solr/core/src/java/org/apache/solr/jersey/PostRequestDecorationFilter.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.solr.jersey;
+
+import org.apache.solr.core.SolrCore;
+import org.apache.solr.request.SolrQueryRequest;
+import org.apache.solr.request.SolrRequestHandler;
+import org.apache.solr.response.SolrQueryResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerResponseContext;
+import javax.ws.rs.container.ContainerResponseFilter;
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+
+import static org.apache.solr.jersey.RequestContextConstants.SOLR_QUERY_REQUEST_KEY;
+
+/**
+ * Applies standard post-processing decorations to a {@link SolrJerseyResponse} that are needed on all responses.
+ *
+ * @see SolrCore#postDecorateResponse(SolrRequestHandler, SolrQueryRequest, SolrQueryResponse)
+ */
+public class PostRequestDecorationFilter implements ContainerResponseFilter {
+
+    private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+    @Override
+    public void filter(ContainerRequestContext requestContext, ContainerResponseContext responseContext) throws IOException {
+        final SolrQueryRequest solrQueryRequest = (SolrQueryRequest) requestContext.getProperty(SOLR_QUERY_REQUEST_KEY);
+        if ( ! responseContext.hasEntity() || ! SolrJerseyResponse.class.isInstance(responseContext.getEntity())) {
+            log.debug("Skipping QTime assignment because response was not a SolrJerseyResponse");
```

Comment:
Would we do QTime if we started from scratch?

Reply:
Do you mean in terms of a field name?  Or it terms of having the field at all?

I don't care for the name.  But having a field to return this info seems reasonable to me?

Reply:
Having the field at all...    I tried to find a nice authoritative article that says "in rest design, QTime (or it's like), is a bad idea", but honestly, I didn't find it.   And it is nice to see which api calls take a long time to run!   So, I guess, this is cool feature of Solr!

Reply:
Yeah, I can't say I've actually looked at the value in the response much.

I use the value recorded in the solr logs frequently, to do performance analysis, etc.  But rarely (if ever) have I written code to pull the value out of individual SolrJ responses and do something with it.  Seems like a larger question though - I'd be curious what (if anything) people use this for.

Reply:
`qtime` is definitely useful for some quick performance checks if you are running some queries. I also know I've seen it be helpful if there are query timeouts just to see how long the result took. Easier than trying to coordinate in the logs. Not sure outside of that.